system_modules/naprender/src/bufferbindinginstance.cpp

```diff
@@ -72,12 +72,33 @@ namespace nap
 	static const ShaderVariableDeclaration& getBufferDeclaration(const BufferObjectDeclaration& declaration)
 	{
 		// If a buffer object declaration is passed, we can safely acquire the actual buffer declaration from it
-		if (declaration.get_type() == RTTI_OF(BufferObjectDeclaration))
+		const auto* buffer_object_declaration = rtti_cast<const BufferObjectDeclaration>(&declaration);
```

Comment:
This call seems to be redundant now? You can call `declaration.getBufferDeclaration()` directly, removing the need for this static function completely?